### PR TITLE
Add vision models to doc tests

### PR DESCRIPTION
**Pull request by NielsRogge**
# What does this PR do?

This PR adds all vision models to the doc tests (and fixes the code examples of `xxxForMaskedImageModeling` models).

Thanks @patrickvonplaten for reviving these.

### CODE REVIEW

File: utils/documentation_tests.txt [hunk 1]
```diff
@@ -7,3 +7,12 @@ src/transformers/models/sew/modeling_sew.py
...
 docs/source/quicktour.mdx
 docs/source/task_summary.mdx
+src/transformers/models/vit/modeling_vit.py
+src/transformers/models/beit/modeling_beit.py
+src/transformers/models/deit/modeling_deit.py
+src/transformers/models/swin/modeling_swin.py
+src/transformers/models/convnext/modeling_convnext.py
+src/transformers/models/poolformer/modeling_poolformer.py
+src/transformers/models/vit_mae/modeling_vit_mae.py
+src/transformers/models/segformer/modeling_segformer.py
```
Comment (sgugger) [inline utils/documentation_tests.txt:17]: Nit, can you put all `src` together and all `docs` together?

**Review by sgugger — APPROVED** [review]
Thanks for fixing those!

File: src/transformers/models/beit/modeling_beit.py [hunk 2]
```diff
@@ -763,9 +764,13 @@ def forward(
...
-        >>> inputs = feature_extractor(images=image, return_tensors="pt")
-        >>> outputs = model(**inputs)
-        >>> logits = outputs.logits
+        >>> num_patches = (model.config.image_size // model.config.patch_size) ** 2
+        >>> pixel_values = feature_extractor(images=image, return_tensors="pt").pixel_values
+        >>> # create random boolean mask of shape (batch_size, num_patches)
+        >>> bool_masked_pos = torch.randint(low=0, high=2, size=(1, num_patches)).bool()
+
+        >>> outputs = model(pixel_values, bool_masked_pos=bool_masked_pos)
+        >>> loss, logits = outputs.loss, outputs.logits
```
Comment (patrickvonplaten) [inline src/transformers/models/beit/modeling_beit.py:773]: should we maybe check for the shape of the logits here? So that we actual test something
